src/defect/DefectService.ts

```diff
@@ -0,0 +1,231 @@
+import moment from 'moment';
+import { Service } from 'typedi';
+import { ICustomDefect } from '../models';
+import { IVA_30, LOCATION_ENGLISH, LOCATION_WELSH } from '../models/Enums';
+import { IDefectChild } from '../models/IDefectChild';
+import { IDefectParent } from '../models/IDefectParent';
+import { IFlatDefect } from '../models/IFlatDefect';
+import { IItem } from '../models/IItem';
+import { TestResultService } from '../test-result/TestResultService';
+
+@Service()
+export class DefectService {
+	constructor(private testResultService: TestResultService) {}
+
+	/**
+	 * Calculates the retest date for an IVA or MSVA test
+	 * @param testTypeStartTimestamp - the test start timestamp of the test
+	 */
+	public calculateVehicleApprovalRetestDate = (testTypeStartTimestamp: string): string => {
```

Comment:
This method is no longer being used as FE now provide the date in the payload, so can be removed. Thanks.

Reply:
Good spot, will remove 🙏